Makefile

```diff
@@ -98,11 +97,19 @@ CCFLAGS=-std=gnu11 -I$(IDIR) $(WARNFLAGS) -D_FILE_OFFSET_BITS=64 $(HARDENING_FLA
 # for FTL we need the pthread library
 # for dnsmasq we need the nettle crypto library and the gmp maths library
 # We link the two libraries statically. Although this increases the binary file size by about 1 MB, it saves about 5 MB of shared libraries and makes deployment easier
-#LIBS=-pthread -lnettle -lgmp -lhogweed
-LIBS=-pthread -lrt -Wl,-Bstatic -L/usr/local/lib -lhogweed -lgmp -lnettle -Wl,-Bdynamic
+LIBS=-pthread -lrt -Wl,-Bstatic -L/usr/local/lib -lhogweed -lgmp -lnettle
 # Flags for compiling with libidn : -lidn
 # Flags for compiling with libidn2: -lidn2
 
+# Do we want to compile a statically linked musl executable?
+ifeq "$(CIRCLE_JOB)" "x86_64-musl"
```

Comment:
Can this be a more general check (not tied to the CI)? Something like checking if `MUSL` equals `true`?

Reply:
Actually, doesn't need to be tied to musl, the variable could be named `STATIC`.

Reply:
I had the same idea initially. For the same of simplicity (use the same build job for all architectures), we'd realize this via an environment variable set in the musl docker container. What do you think about this?

Reply:
To make it easier to change the value, we should instead set the variable in the CircleCI environment variables for that job.

Reply:
I intentionally didn't choose this path as it again ties us more to the CI. The sole purpose of the `musl` container is to provide a self-contained binary. The container should behave similarly on the CI and when compiling local. This wouldn't be the case if we add in a (hidden) extra environment variable in the CI.